SCUMM: Add support for Catalan SCUMM games

Jibun opened the pull request:
- Add MM Catalan font and special characters
- Add MM Catalan prepositions
- Add MD5 detection entries for Catalan translation of MM

dwatteau commented:
@Jibun: Thanks for submitting this PR!

(The translation patch is [hosted here](https://comunitat.cetrencada.cat/?entrada=scumm-mm), for though looking to test it.)

Your PR mostly looks good to me; a few remarks before it could be merged, though:

1. FWIW, the patch they provide for the V2/Enhanced version is not compatible with the "real" English version. It looks like it's only compatible with the Steam/GOG release (which is actually a weirdly cracked version that Disney provides nowadays). The "real" Enhanced/V2 English release is found in the Day of the Tentacle Remaster file, though, using [DoubleFine Explorer](https://quickandeasysoftware.net/software/doublefine-explorer) to extract its `maniac/` content.
   * This is not a blocker, of course; the translation patch just seems to require the Steam/GOG release, so maybe they want to provide some `.ips` file that'd also be compatible with the different `00.LFL` file the original release used (in case you're part of the team which produced this translation).
2. I think this PR is missing the translated strings for the Pause bar and such. For V1/V2 games, as far as I can remember it's hardcoded in the original `MANIAC.EXE` file, so one has to use the patched executable they provide and use DOSBox to see them in-game.
  * [Here's where](https://github.com/scummvm/scummvm/blob/cb830ebc467cb2a4d17596382356ddea5a4fb15e/engines/scumm/dialogs.cpp#L582-L643) it's handled in ScummVM.

So far, I've located some of those strings:

```
"No es troba l'arxiu nn.lfl Prem ESPAI"
"Joc pausat, prem ESPAI per continuar."
"Estas segur que vols reiniciar? (y>n)"
```

but I couldn't find them all, and also we'd need a translated variant of "Estas segur que vols reiniciar" where "reiniciar" would be replaced by the Catalan translation of "quit".

Could you either add this `InfoDialog::getStaticResString()` change in your PR, or just help figure out the missing strings for that? Thanks!

Jibun commented:
A quick question: I need to add the missing Japanese entries to `strMap1` within `InfoDialog::getStaticResString()` so as not to mess up the language index used later in the same function (Japanese is 7, Catalan will be 8).
Should I leave them empty or copy the English entries?

Jibun commented:
> BTW: For patching the resources, did you use scummtr and/or some other tool? I never tried to translate any V1/V2 game myself, but as far as I can remember, scummtr had some reliability issues with v2 games.

I used scummtr as well as several other tools, some tweaked by me since they didn’t work well or didn’t provide what was needed. However, it’s a bit blurry now.
Regarding scummtr, I know I couldn’t use your version (just found out the repo was yours), since it doesn’t work well with MM. I used the original 0.4.0 version instead.
If you’d like more concrete info, I can look into it and let you know the details.

bluegr commented:
> A quick question: I need to add the missing Japanese entries to `strMap1` within `InfoDialog::getStaticResString()` so as not to mess up the language index used later in the same function (Japanese is 7, Catalan will be 8). Should I leave them empty or copy the English entries?

Yes, just add the English strings with a TODO to translate them in the future

Jibun commented:
Hey!
I was testing that all the menus were working correctly in the other SCUMM games we’ve translated at Ce Trencada before adding their md5, and I found that _Indy FoA_ (DOS CD version) crashes when entering the save menu. The English version also crashes.
I've tried with other games, and it seems the same happens with several versions of tSoMI, just to mention one, the DOS CD English version.

I don’t know if it’s something with my local setup, but I don't see how. It doesn’t happen with the latest ScummVM release, though, so it may be an issue introduced in the current version. Could you try it on your side?

The error seems to be that, in _dialogs.cpp_, a weird number is passed to `stringno` in the `InfoDialog::getStaticResString` function, which causes the assert at line 851 to fail.

dwatteau commented:
Hey!

> [...] I found that _Indy FoA_ (DOS CD version) crashes when entering the save menu. The English version also crashes. I've tried with other games, and it seems the same happens with several versions of _tSoMI_, just to mention one, the DOS CD English version.
>
> I don’t know if it’s something with my local setup, but I don't see how. It doesn’t happen with the latest ScummVM release, though, so it may be an issue introduced in the current version. Could you try it on your side?

I'm afraid I'm not seeing any crash with Monkey1 or FOA. Monkey1 is the VGA one with the inventory icons, and FOA is the English talkie one. Then, in the Game settings, the "Enable the original GUI and Menu" is turned on; is that what you're testing as well?

Also, what is your environment? GCC, Clang, MSVC?

(FWIW, I'm running Clang on macOS on my side, with UBSan and ASan turned on to catch a few more common bugs, but so far I couldn't trigger any crash here.)
 
> The error seems to be that, in _dialogs.cpp_, a weird number is passed to `stringno` in the `InfoDialog::getStaticResString` function, which causes the assert at line 851 to fail.

I'm not seeing an assert around line 851 of `dialogs.cpp`, neither in the current Git HEAD nor in previous v2.9.1 or v2.9.0 tags 🤔😅

If you can build with debug symbols on your side, do you have a way of showing a full backtrace of the crash you're experiencing?

(and don't hesitate coming over Discord if it's easier for you to debug things there)

bluegr commented:
I've just tried the English versions of FoA CD and MI1 CD, with both the original and ScummVM save/load menus, and I don't get any crashes - everything works as expected. Tried with both the stable version 2.9.1 and the latest daily version. I'm on Windows 11, using MSVC 2022.

So, it must be something with your setup @Jibun

Jibun commented:
Ok, sorry for the delay in updating this PR.
I’ve now updated it to include all the available SCUMM Catalan games.
I’ve also added the missing hardcoded dialogs and fixed a hidden issue we found in `dialogs.cpp`, which I discussed with @dwatteau on Discord and resolved with his help 🙏🏻.

@bluegr thank you very much for checking it, and sorry for not updating the information here sooner.

AndywinXp commented:
Hmm, the fix seems theoretically correct. But it breaks other versions (e.g. the english version of Maniac Mansion)

dwatteau reviewed (approved):
LGTM

Jibun commented:
Thank you all for the given support!
